### PR TITLE
fix(vue-app): reload page once after loading chunk error

### LINKED ISSUE #3389: [edge] Deployment of new version breaking site for live users. ( Loading chunk {n} failed error)
### Version

[v1.4.0](https://github.com/nuxt/nuxt.js/releases/tag/v1.4.0)

### Reproduction link

[None](None)

### Steps to reproduce

This  happens on a live site, production deployed not a dev.  The sequence of events following:

User opened the site home page
New version of the app was deployed to web servers
User clicked a link to another page within the nuxt SPA
User saw Loading chunk {n} failed message instead of the new page content
https://i.imgur.com/AO7gX9h.png

Any site deploy could break the SPA for  live users and cause them to need a refresh.

This should be handled properly by nuxt, maybe something like automatically refresh in such cases. This is just really frustrating and bad behavior for users

### What is expected ?

No error displayed to user. Site autoreload or whatever will work here

### What is actually happening?

User see some cryptic error 

Afaik nuxt 1.4 have same behaviour 

<div align="right"><sub><em>This bug report is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt/nuxt.js/issues/c7171">#c7171</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by rinu**
## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description
Resolves: #3389
Changes made in #3940 should maybe be deleted.
There's probably a nicer way of doing this, all comments are welcome.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Review by clarkdo — DISMISSED** [review]

File: packages/vue-app/template/utils.js [hunk 1]
```diff
@@ -146,7 +146,19 @@ export function resolveRouteComponents (route, fn) {
...
       if (typeof Component === 'function' && !Component.options) {
-        Component = await Component()
+        try {
+          Component = await Component()
+        } catch (error) {
+          // Handle webpack chunk loading errors
+          // This may be due to a new deployment or a network problem
+          if (error && error.name === 'ChunkLoadError' && window.location.hash !== '#retry') {
+            // mark the page not to reload infinitely
+            window.location.hash = '#retry'
```
Comment (pi0) [inline packages/vue-app/template/utils.js:156]: Modification of URL, might interference with application logic. We may use [navigation type](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigation/type) guard. (safari workaround needed)

Comment (pi0) [inline packages/vue-app/template/utils.js:156]: Modification of URL, might interference with application logic. We may use [navigation type](https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigation/type) guard. (safari workaround needed)

**Review by pi0 — CHANGES_REQUESTED** [review]

File: packages/vue-app/template/utils.js (hunk 1 above)
Comment (rinu) [reply packages/vue-app/template/utils.js:156]: There's no clean way to solve this, can't think of anything at least. Navigation type seems unreliable. We could rename this retry to #__nuxt_retry or something. Or try to use sessionStorage for this?

We could not have this check at all but there's the unlikely possibility of a currupt build where a critical file is actually missing from the file system. Infinite reloads will not help when debugging that.

Comment (pi0) [reply packages/vue-app/template/utils.js:156]: Do you have any reference regarding the unreliability of nav type? But of course we can detect incompatible UA and switch to localStorage (or sessionStorage) method.   Modification of `#` overrides one if set in real application.

Comment (rinu) [reply packages/vue-app/template/utils.js:156]: As you pointed out navigation type is not supported by Safari so I can't rely on it. I'm working on a sessionStorage method.

**Review by pi0 — APPROVED** [review]

**Comment by pi0** [comment]
Thanks :)

**Pull request merged**